session.go

```diff
@@ -647,6 +707,19 @@ func (s *session) handleSinglePacket(p *receivedPacket, hdr *wire.Header) bool /
 		packet.hdr.Log(s.logger)
 	}
 
+	if s.mitigationOn(packet.encryptionLevel) {
+		s.saveState()
+		if err := s.handleUnpackedPacket(packet, p.rcvTime); err != nil {
+			s.logger.Debugf("Ignoring or trying to recover from suspicious/invalid packet, rolling back relevant state.")
+			s.restoreState()
```

Comment:
What kind of packet would both (1) cause a connection ID change and (2) cause an error here?  An Initial containing a CONNECTION_CLOSE with a different connection ID?

Could we just ignore the server connection ID change, if the packet closes the connection anyway?